GreengrassIPC.md

```diff
@@ -0,0 +1,172 @@
+public API.
+
+## Cross-SDK Differences
+The Javascript Greengrass IPC implementation has several differences relative to the IPC implementations in the other v2 SDKs:
+* No handler types - Javascript Greengrass IPC uses Javascript's event system (events, listeners, emitters) for all asynchronous notifications.
+* Single client - Other SDKs have both a v1 and v2 client, where the v1 client is low-level and more verbose (and complex) in usage, while the v2 client is streamlined and less prone to accidental misuse.  Javascript Greengrass IPC skips the v1 API and starts directly with the v2 client approach.
+
+## Greengrass IPC Example
+For those who prefer to dive right in, you can start with our [Greengrass IPC example](./samples/node/gg_pic/index.ts).  
+The associated [README](./samples/node/gg_ipc/README.md) contains instructions for how to perform a local greengrass deployment of the sample
+as a component.
+
+## Greengrass IPC Client
+All Greengrass IPC is initiated from a Greengrass IPC Client.  A single client should be sufficient for all use cases,
+regardless of concurrency requirements.
+
+### Client Setup
+
+While advanced configuration options exist, under typical circumstances creating a Greengrass IPC client is quite easy:
+
+```typescript
+import * as greengrascoreipc from 'aws-iot-device-sdk-v2';
+
+let client = greengrascoreipc.createClient();
+```
+
+Once created, call the `connect` API to establish a connection from your component to its Greengrass Nucleus:
+```typescript
+await client.connect();
+```
+
+`connect` may only be called once.  If a client becomes disconnected and you wish to continue to use Greengrass IPC, you must
+create a new client.  If `connect` fails, the associated Promise will be rejected with an 
+[RpcError](https://aws.github.io/aws-iot-device-sdk-js-v2/node//classes/eventstream_rpc.RpcError.html) value.  See 
+[Client Error Handling](#Client-Error-Handling) for more information on `RpcError`.
+
+### Client Shutdown
+When finished with the client, you must call the `close` API, regardless of whether the network connection is still active.
+`close` also releases native resources used by the client, and without invoking it, they will leak:
+
+```typescript
+await client.close();
+```
+
+As part of the asynchronous `close` process, all unclosed streaming operations associated with the client will also
+be closed.
+
+Once a client has been closed, it cannot be used further.
+
+### Client Events
+The Greengrass IPC client provides a single event that can be listened to, the `disconnection` event.  A `disconnection`
+is emitted when the underlying connection is torn down.  A `disconnection` event will only be emitted if a call to 
+`connect` previously succeeded; connection establishment failures only manifest as a rejected promise.
+
+Calling `close` on a connected client will cause a `disconnection` event to be emitted as part of the shutdown process.
```

Comment:
Question: Is there a way to tell the difference between a disconnection event caused by calling `close` versus a disconnection event caused by the client becoming disconnected from Greengrass IPC?

Reply:
The associated CrtError will be "user-initiated disconnect"